src/runtime/internal/utils.ts

```diff
@@ -189,3 +189,14 @@ export const has_prop = (obj, prop) => Object.prototype.hasOwnProperty.call(obj,
 export function action_destroyer(action_result) {
 	return action_result && is_function(action_result.destroy) ? action_result.destroy : noop;
 }
+
+export function split_css_unit(value: number | string): [number, string] {
+	if (typeof value === 'number') {
+		return [value, 'px'];
+	}
+	const split = value?.match?.(/^\s*(-?[\d.]+)([^\s]*)\s*$/);
+	if (split) {
+		return [parseFloat(split[1]), split[2] || 'px'];
+	}
+	return [parseFloat(value), 'px'];
+}
```

Comment:
if we're concerned about the size of this function, I *think* the following would be equivalent and a bit smaller
```suggestion
export function split_css_unit(value: number | string): [number, string] {
	const split = typeof value === 'string' && value?.match?.(/^\s*(-?[\d.]+)([^\s]*)\s*$/);
	return split ? [parseFloat(split[1]), split[2] || 'px'] : [value, 'px'];
}
```